src/libopensc/card-openpgp.c

```diff
@@ -787,6 +788,14 @@ pgp_enumerate_blob(sc_card_t *card, struct blob *blob)
 		}
 		tag |= cla;
 
+		/* Awful hack for composite DOs that have
+		 * a TLV with the DO's id encompassing the
+		 * entire blob. Example: Yubikey Neo */
+		if (tag == blob->id) {
```

Comment:
A quick reading of the GnuPG code (http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=common/tlv.c;h=c6875640690c8410406d7b19afebcd543bb42c65;hb=refs/heads/STABLE-BRANCH-2-0#l39) seems to imply that it is willing to recurse up to 100 times when looking in a constructed DO. This would imply that the above side effect isn't too major because it seems GnuPG would accept that anyways.